### ISSUE TITLE
Using QR Codes to track hacker activity

### ISSUE THREAD

**Issue by SaltyQuetzals** [techx/quill#64]
There's a lot of untapped data in hackathons that can be utilized if, in addition to (or instead of) using paper IDs distributed at the beginning of the event, users had QR codes that could be scanned. This could help determine which events are popular among hackers, which sponsors are the most popular (and should return the next year), who has had their meals and who hasn't, and, depending on how in-depth you want to be, you can get data on all kinds of activities, determining which snacks are the most popular, etc.

## Event Popularity Tracking
At a mini event (say, a typing competition) supervising organizers/staff members open a link on Quill that requests access to their camera. Once granted, they are able to scan hackers who want to participate. This tracks the popularity of the typing competition, and indicates whether it should make an appearance next year.

## Sponsor Popularity Tracking
Similarly to the example above, a sponsor representative could open a link that requests access to their camera, and once granted, they're able to scan QR codes of hacker who are interested in networking or just making conversation. This is a mutually-beneficial situation as well, as sponsors could maybe view the hacker's resume after scanning the code, and the event organizers can track which sponsors are popular and should make a reappearance.

## Meal Tracking
Meal tracking is an interesting application of this system. Rather than checking boxes on a paper ID (which costs money to produce), users could have their codes scanned when getting a meal, which "checks" their meal off. Another application of the code scanning would be optimizing dietary restriction ordering. If 20 people indicate that they want a vegetarian option, and 30 people over the course of the event scan in at the vegetarian line, maybe next year, more vegetarian meals could be ordered.

Some people might have qualms with their data being tracked, so it could be an opt-in system.

**Comment by jlin816** [comment]
Yes! We actually did this with RFID badges / USB scanners for HackMIT '17. Tagging @revalo @patins @ajayjain who worked on it, and might have the counting and tracking software we used. If I remember correctly it was linked to people's registration information from Quill. That might be an easy out-of-the-box solution for you or anyone who wants to implement something like this at their event without having to build an entirely new feature in Quill (which I'm not entirely convinced is the right place for this to go, anyway).
